### ISSUE TITLE
Update Dependencies

### ISSUE THREAD

**Issue by balupton** [docpad/docpad#1045]
Work is commencing over at: https://github.com/docpad/docpad/tree/dev-updatedeps

Primary todos:
- [x] Update deps
  - [x] Have core tests pass
  - [x] Have plugin tests pass
- [x] Removed `--profile` as better ways of doing that these days
  - [x] Remove lazy-require dep, as may no longer be needed, as was mostly used for above removed use case
- [x] Added babel polyfill as most deps now use node 0.12 minimum

Secondary todos:
- [x] Update watchr to have latest deps
- [x] Update bal-util to have latest deps
- [x] Update docpad for scandir-cluster
- [x] Add npm deprecation notice to scandirectory recommending scandir-cluster

Failing tests:
- Nested configuration test is failing, could be an extendr regression. Here is the failed test https://travis-ci.org/docpad/docpad/builds/137858315 - if anyone wants to help, please do - not sure when I will get back around to this

**Comment by balupton** [comment]
Also removed `--profile` as better ways of doing that these days

**Comment by balupton** [comment]
Also added the babel polyfill

**Comment by SteveMcArthur** [comment]
I pushed up some changes and test now passing. Looks like there was a problem with extendr as you suggested. Wound back some of the dependency updates - but now that the tests are passing we can start adding them back in.

**Comment by SteveMcArthur** [comment]
OK - dependency updates reapplied - except event-emitter-grouped which is fixed at 2.4.3, joe at 1.6.0 and extendr at ^2.1.0

**Comment by SteveMcArthur** [comment]
Express dependency not updated to v4. I'm guessing this is a separate piece of work as migrating to v4 is a massive breaking dependency update.

**Comment by SteveMcArthur** [comment]
lazy-require removed

**Comment by SteveMcArthur** [comment]
@balupton I think I've solved most of the problems. Mostly waiting on solutions to the joe/event-emitter-grouped issues and the extendr issue. Not sure where we are with express.

**Comment by SteveMcArthur** [comment]
@balupton I'm thinking we should apply the dependency updates as is and deal with the remaining problems separately.

**Comment by balupton** [comment]
> @balupton I'm thinking we should apply the dependency updates as is and deal with the remaining problems separately.

this issue should be seperate from #1047

**Comment by SteveMcArthur** [comment]
@balupton no worries - just an idle thought really. #1047 is already applied to master. :)

**Comment by balupton** [comment]
@SteveMcArthur not really sure then what https://github.com/docpad/docpad/compare/1a4aece023726c13beba12be801c0d0bfb2485f2...b9d9bb79cd2cbb9cf451e5cf069a340d112aa639 is meant to do, besides downgrade deps, which isn't what this branch is about

**Comment by SteveMcArthur** [comment]
@balupton The idea really was to isolate what was causing the tests to fail. So have all deps upgraded that pass applied. The remaining deps are awaiting issues on the third party deps. Once these are solved then the remaining deps can be reapplied.

**Comment by SteveMcArthur** [comment]
@balupton Pushed up extendr dep. Possibly an issue with CSON.

**Comment by balupton** [comment]
Issues so far:

---

```
Comparison Diff:
{ a: 'instanceConfig', b: 'instanceConfig', c: 'websiteConfig' }{ a: 'websiteConfig', b: 'websiteConfig', c: 'websiteConfig' }

Comparison Actual:
{ a: 'websiteConfig', b: 'websiteConfig', c: 'websiteConfig' }

Comparison Expected:
{ a: 'instanceConfig', b: 'instanceConfig', c: 'websiteConfig' }
```

Due to:

> Configuration test was not actually correct. Fixed the test and changed the code to deep extend, without safety/default consideration.

---

```
docpad-actions ➞  generate ➞  action
Error: extendr only supports extending plain objects, an input was not a plain object
  at custom (/Users/balupton/Projects/active/docpad/node_modules/extendr/source/index.js:22:10)
  at Object.extend (/Users/balupton/Projects/active/docpad/node_modules/extendr/source/index.js:68:9)
  at /Users/balupton/Projects/active/docpad/out/lib/models/document.js:217:25
  at DocumentModel.FileModel.parse (/Users/balupton/Projects/active/docpad/out/lib/models/file.js:1076:5)
  at DocumentModel.parse (/Users/balupton/Projects/active/docpad/out/lib/models/document.js:172:35)
  at Task.<anonymous> (/Users/balupton/Projects/active/docpad/out/lib/models/file.js:1006:19)
  at ambi (/Users/balupton/Projects/active/docpad/node_modules/ambi/source/index.js:84:3)
  at Domain.fireMethod (/Users/balupton/Projects/active/docpad/node_modules/taskgroup/source/lib/task.js:519:5)
  at Domain.run (domain.js:221:14)
  at Task.fire (/Users/balupton/Projects/active/docpad/node_modules/taskgroup/source/lib/task.js:537:15)
  at Immediate.queue (/Users/balupton/Projects/active/docpad/node_modules/taskgroup/source/lib/task.js:572:20)
  at runCallback (timers.js:566:20)
  at tryOnImmediate (timers.js:546:5)
  at processImmediate [as _immediateCallback] (timers.js:525:5)
```

Due to:

```
> require('CSON').parseString(a, {json:true, cson:true, coffeescript: true, javascript: true, format: 'coffeescript'}).__proto__ === Object.prototype
false
```

**Comment by balupton** [comment]
@SteveMcArthur I'm resuming work on this - so dropped your changes

**Comment by balupton** [comment]
As this is now finishing up, https://github.com/docpad/docpad/issues/1046 is the next item

**Comment by balupton** [comment]
Turns out https://github.com/docpad/extras/tree/docpad-6.x doesn't even run anymore. Converting it to latest deps and to javascript.

**Comment by balupton** [comment]
all supported plugins have now been updated to latest base files, and docpad/extras has now been updated accordingly

this means I can now test upcoming docpad releases against all our supported plugins accordingly, and as such, test the dependency update branch against them, in order to do the next 6.x release!

**Comment by balupton** [comment]
Pushed to the `beta` git branch, and released to `6.80.0-beta`, under the `beta` npm tag
